libyul/AsmPrinter.cpp

```diff
@@ -212,15 +216,19 @@ string AsmPrinter::operator()(Switch const& _switch) const
 string AsmPrinter::operator()(ForLoop const& _forLoop) const
 {
 	solAssert(_forLoop.condition, "Invalid for loop condition.");
-	string out = "for ";
-	out += (*this)(_forLoop.pre);
-	out += "\n";
-	out += boost::apply_visitor(*this, *_forLoop.condition);
-	out += "\n";
-	out += (*this)(_forLoop.post);
-	out += "\n";
-	out += (*this)(_forLoop.body);
-	return out;
+	string pre = (*this)(_forLoop.pre);
+	string condition = boost::apply_visitor(*this, *_forLoop.condition);
+	string post = (*this)(_forLoop.post);
+	char delim = '\n';
+	if (
+		pre.size() + condition.size() + post.size() < 60 &&
+		pre.find('\n') == string::npos &&
+		post.find('\n') == string::npos
+	)
+		delim = ' ';
+	return
+		("for " + move(pre) + delim + move(condition) + delim + move(post) + "\n") +
```

Comment:
Moving doesn't help for concatenation at all - it needs to be copied anyways, right?
One could also think about benchmarking string concatenation - I reckon that a custom concatenation function that creates an empty string, ``reserve()``s the required space and then adds the elements with ``append`` will be fastest, but that's independent of this PR.

Reply:
There are rvalue overloads for `operator+` for strings. It migh not make too much sense in this case, but `move(x) + move(y)` is certainly faster than `x+y` in the case that `x` does not have to be resized.

I benchmarked the use of `reserve` some years back and could not find any speed benefit.

Reply:
Ok - well the move doesn't hurt in any case and maybe ``"for " + move(pre)`` will even use ``pre``s space if that suffices.
I just read some  benchmarks suggesting that ``reserve`` is fastest, though - maybe one should check that again - but in any case nothing for this PR.